### PR TITLE
Add IBS Mirror peering in qesap regression Azure

### PR DESCRIPTION
**Pull request by mpagot**
Add network peering in qesap regression for Azure. Minor fixes in qesapdeployment and unit testing about peering functions.

Related ticket: [TEAM-7704](https://jira.suse.com/browse/TEAM-7704)

Verification run: Azure 15sp4
- saptune http://openqaworker15.qa.suse.cz/tests/153896#step/configure/85 (without QESAP_IBSMIRROR_RESOURCE_GROUP) generated .tfvars is http://openqaworker15.qa.suse.cz/tests/153896/file/configure-terraform.tfvars

```
vnet_address_range = "10.2.64.0/21"
subnet_address_range = "10.2.64.0/24"
```
`az` commands for peering are not called due to missing **QESAP_IBSMIRROR_RESOURCE_GROUP** setting

- saptune http://openqaworker15.qa.suse.cz/tests/153897 (with QESAP_IBSMIRROR_RESOURCE_GROUP) generated .tfvars is http://openqaworker15.qa.suse.cz/tests/153897/file/configure-terraform.tfvars

```
vnet_address_range = "10.2.88.0/21"
subnet_address_range = "10.2.88.0/24"
```

The `az` peering create command is in http://openqaworker15.qa.suse.cz/tests/153897#step/test_cluster/346 and it is fine

The peering delete one is in http://openqaworker15.qa.suse.cz/tests/153897#step/test_cluster/372  and it fails for `timed out`

- saptune with QESAP_IBSMIRROR_RESOURCE_GROUP and longer `az` timeout http://openqaworker15.qa.suse.cz/tests/153916

- sapconf http://openqaworker15.qa.suse.cz/tests/153901#live (this is using the config.yaml not edited by this PR)

One more set of VR after the alignment to the master content:

saptune without QESAP_IBSMIRROR_RESOURCE_GROUP :: clone of VR 153896 ---> http://openqaworker15.qa.suse.cz/tests/156435

saptune with QESAP_IBSMIRROR_RESOURCE_GROUP and longer timeout :: clone of VR 153916 ---> http://openqaworker15.qa.suse.cz/tests/156435

sapconf :: clone of VR 153901 ---> http://openqaworker15.qa.suse.cz/tests/156435

### CODE REVIEW

File: data/sles4sap/qe_sap_deployment/qesap_azure.yaml
Comment (BillAnastasiadis) [inline data/sles4sap/qe_sap_deployment/qesap_azure.yaml:13]: LGTM, my only question is, do we want `vnet_address_range` and `subnet_address_range` vars defined in the main yaml, or we want a new yaml schedule specifically for maintenance jobs that will include them? 

I'm asking in case we want to have runs using `qesap_azure_yaml` that won't need the mirror, and thus won't need peering and address ranges as well

File: tests/sles4sap/qesapdeployment/test_cluster.pm
Comment (alvarocarvajald) [inline tests/sles4sap/qesapdeployment/test_cluster.pm:32]: Is this OK? Delete peering right after setting it up?

Comment (mpagot) [reply tests/sles4sap/qesapdeployment/test_cluster.pm:32]: it is intentionally the bare minimal test to validate the new API

Comment (BillAnastasiadis) [reply tests/sles4sap/qesapdeployment/test_cluster.pm:32]: I guess this is a test setting, i.e. tries to test the peering without actually applying any updates.

However, we probably shouldn't merge it if it's only for testing? As far as I understand, the peering will happen either in a separate module or during the module that adds the repos, probably not in `test_cluster` (but Michele may have another idea)

Comment (mpagot) [reply tests/sles4sap/qesapdeployment/test_cluster.pm:32]: Why not to merge this even if the peering is only to test the API? As we are using the WORKER_ID as seed we will not "steal" any resources (like IP ranges).

Comment (mpagot) [reply tests/sles4sap/qesapdeployment/test_cluster.pm:32]: Why not to merge this even if the peering is only to test the API? As we are using the WORKER_ID as seed we will not "steal" any resources (like IP ranges). The purpose of this is really only to probe the peering create and destroy AZ commands, the `vnet_address_range` and `subnet_address_range` settings in qe-sap-deployment and eventually partially the idea about using the WORKER_ID as seed

Comment (BillAnastasiadis) [reply tests/sles4sap/qesapdeployment/test_cluster.pm:32]: True, one would say that this may even be a part of testing the cluster, so it belongs here anyway. Fine to me

**Review by BillAnastasiadis — APPROVED** [review]
LGTM.

**Comment by mpagot** [comment]
Obsoleted by https://github.com/os-autoinst/os-autoinst-distri-opensuse/pull/17048

**Pull request closed without merging**
